includes/functions/functions_products.php

```diff
@@ -338,11 +338,8 @@ function zen_get_linked_products_for_category($category_id, $first_only = false)
     $results = $db->Execute($sql);
 
     if ($first_only) {
-        if ($results->RecordCount()) {
-            return $results->fields['products_id'];
+        return $results->RecordCount() ? (int)$results->fields['products_id'] : 0;
```

Comment:
i am not a fan of the ternary operator in this situation.  but i suppose a question of style.  in addition, there is no need for any casting to `(int)`.  your original PR needed to do that when an empty string was returned.  we are now returning `0` so no casting is necesssary, as `products_id` is defined as an integer.

personally i prefer:
```
if ($results->EOF) {
    return 0;
}
return $results->fields['products_id'];
```
`EOF` is already set on the `Execute`, while `RecordCount()` is a function.  the speed difference is probably neglible due to `RecordCount()` being cached.  but, hey, its your PR.